packages/codemods/documentation/howTo.md

```diff
@@ -0,0 +1,119 @@
+- **Renaming/Repathing imports**: Given a search string / regex, an old name/path, and a new name/path, modify all matching import statements to reflect your desired changes!
+
+## Writing Your First CodeMod
+
+Let's say you were responsible for renaming the `toggled` prop in Fluent's `Dropdown` component to `checked`. There are **two** easy ways to do it!
+
+Feel free to try this out on your local branch of fluent! I've set up the files for you :).
+
+### JSON
+
+- To create a codemod using the config file, all you'd need to do is go into the `upgrades.json` file and replace what's there with the following template as defined in `src/codemods/types.ts`:
+
+```jsonld=
+{
+  "name": // Classify this collection of codemods
+  "upgrades": [
+    {
+      "name": // What upgrade are you running?
+      "type": // Currently renameProp | repathImport
+      "version": // Mod version, as a string
+      "options": {
+        "from": {
+          "importName": // Name of component housing prop
+          "toRename": // Prop name to change
+        },
+        "to": {
+          "replacementName": // New prop name
+        }
+      }
+    }
+  ]
+}
+```
+
+- Once you fill out this file, you can then go to [src/codemods/tests/configMod](../src/codemods/tests/configMod/configMod.test.ts) and test your codemod! Invoke the function `createCodeModFromJson()`, which will turn whatever exists in `upgrades.json` into a list of codemods. Check out ##Testing Your Codemods## for more info on how to verify it works!
+
+### Custom CodeMods
+
+- For developers who want a little more flexibility in their mod writing, you can create a codemod manually using codemod _utilities_. Details on these utilities can be found in `src/codemods/utilities`. To rename a prop, we'll need to use two utility functions:
+  - `findJsxTag(file: SourceFile, tag: string)`
+    - This utility returns an array of JSX elements containing the given tag (in our case, the 'tag' will be the name of the component we're looking for).
+    - To create the codemod, you won't need to know much about AST's and how we parse them, but if you want to read up on it, check out [ts-morph!](https://ts-morph.com/)
+  - `renameProp( instances: (JsxOpeningElement | JsxSelfClosingElement)[], toRename: string, replacementName: string, replacementValue?: string, transform?: PropTransform)` \* This function accepts the return value of `findJSXTag`, as well as names for the deprecated prop and its replacement name. The function also takes in an optional replacement value, if you plan on updating the value of this prop whenever you want to rename it. For developers wanting even more granularity with value changes, there's an optional transform function you can pass in -- read[renamePropTransforms](./renamePropTransforms) for more.
+- To create the codemod, wrap these two function calls in a void function that takes in a single parameter of type `SourceFile`. This example suffices:
```

Comment:
We might want to add a section about good practices, like returning a meaningful codemodresult so it can be processed and logged nicely.

Reply:
So you're specific codemod function should return an `Ok({logs: ["useful description of changes that took place"]})` or an `Err({reason: "No operation taken because <Error> or nothing needs to get changed"})`